mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java

```diff
@@ -355,16 +355,16 @@ public Flags getFlags() {
      *            describes which optional data should be returned
      * @return metadata view filtered for the session's user, not null
      */
-    MailboxMetaData getMetaData(boolean resetRecent, MailboxSession mailboxSession, MailboxMetaData.FetchGroup fetchGroup) throws MailboxException;
+    MailboxContentMetaData getMetaData(boolean resetRecent, MailboxSession mailboxSession, MailboxContentMetaData.FetchGroup fetchGroup) throws MailboxException;
 
     /**
      * Meta data about the current state of the mailbox.
      */
-    class MailboxMetaData {
+    class MailboxContentMetaData {
```

Comment:
> Rename to better distinguish the two kind of metadata:
>
> **MailboxMetaData** only includes:
>
> - Path
> - Id
> - ACL
> - HasChildren
> - Counters
>
> **MailboxContentMetaData** adds to this list:
>
> - first unseen
> - recent uids
> - highest modseq
> - permanent flags

These two slightly different names won't help me when reading the code: they mean the same thing for me, they give information about the mailbox.

We need several POJOs to help loading only the right subset of information required but we should not try to find subtle names to distinguish them. Probably the context of use is easier to grasp:

"it's the Metadata from MessageManager"

Or if it's still a problem, maybe grouping them in MailboxMetadata namespace (an empty interface in Java) and give them nice names like: `Short`, `Full`, etc. In APIs, we would then read 'MailboxMetadata.Full`.

WDYT?

Reply:
The idea is nice but the relationship is not as simple: "content" metadata do not posess "HasChild" information.

Reply:
Doing such a split is costly, I won't handle it.

Reply:
If we can't name things, let's keep the same name for all concepts and scope them to their context of use